asyncio task exception turning on switch.rest entity

**Home Assistant release (`hass --version`):**
0.35.0.dev0 (current dev branch)

**Python release (`python3 --version`):**
Python 3.4.2

**Component/platform:**
switch.rest

**Description of problem:**
Every time I call switch.turn_on on a switch.rest entity (which is already on), I get an error message "Error doing job: Task exception was never retrieved", and a stack trace from a failed assertion in asyncio apparently related to the turn_on service call. This is not specific to the automation; I can trigger it with a manual API call to switch.turn_on.

From the log:
```bash
16-12-21 00:34:31 homeassistant.core: Bus:Handling <Event call_service[L]: service_call_id=3051317584-5, service=turn_on, service_data=entity_id=['switch.remote_temperature'], domain=switch>
16-12-21 00:34:31 homeassistant.core: Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/usr/lib/python3.4/asyncio/tasks.py", line 235, in _step
    result = coro.send(value)
  File "/usr/local/lib/python3.4/dist-packages/homeassistant-0.35.0.dev0-py3.4.egg/homeassistant/core.py", line 1042, in _event_to_service_call
    yield from service_handler.func(service_call)
  File "/usr/local/lib/python3.4/dist-packages/homeassistant-0.35.0.dev0-py3.4.egg/homeassistant/components/switch/__init__.py", line 101, in async_handle_switch_service
    update_tasks.append(hass.loop.create_task(update_coro))
  File "/usr/lib/python3.4/asyncio/base_events.py", line 180, in create_task
    task = tasks.Task(coro, loop=self)
  File "/usr/lib/python3.4/asyncio/tasks.py", line 67, in __init__
    assert coroutines.iscoroutine(coro), repr(coro)  # Not a coroutine function!
AssertionError: <Task pending coro=<async_update_ha_state() running at /usr/local/lib/python3.4/dist-packages/homeassistant-0.35.0.dev0-py3.4.egg/homeassistant/helpers/entity.py:191>>
```

**Expected:**
This seems to be a regression from 0.34.2, which was the previous release I was running. This part of the configuration has not changed.

**Problem-relevant `configuration.yaml` entries and steps to reproduce:**
```yaml
switch:
  - platform: rest
    name: Remote Temperature
    resource: 'http://thermostat/tstat/remote_temp'
    body_on: '{"rem_temp":{{ states("sensor.temperature") }}}'
    body_off: '{"rem_mode":0}'
    is_on_template: '{{ value_json.rem_mode != 0 }}'
automation:
  - alias: Send remote temp to thermostat
    initial_state: True
    trigger:
      platform: state
      entity_id: sensor.temperature
    action:
      - service: switch.turn_on
        entity_id: switch.remote_temperature
```

1. Define switch.rest entity
2. Turn it on multiple times via switch.turn_on on the entity.
3. Check log

**Traceback (if applicable):**
```bash
Traceback (most recent call last):
  File "/usr/lib/python3.4/asyncio/tasks.py", line 235, in _step
    result = coro.send(value)
  File "/usr/local/lib/python3.4/dist-packages/homeassistant-0.35.0.dev0-py3.4.egg/homeassistant/core.py", line 1042, in _event_to_service_call
    yield from service_handler.func(service_call)
  File "/usr/local/lib/python3.4/dist-packages/homeassistant-0.35.0.dev0-py3.4.egg/homeassistant/components/switch/__init__.py", line 101, in async_handle_switch_service
    update_tasks.append(hass.loop.create_task(update_coro))
  File "/usr/lib/python3.4/asyncio/base_events.py", line 180, in create_task
    task = tasks.Task(coro, loop=self)
  File "/usr/lib/python3.4/asyncio/tasks.py", line 67, in __init__
    assert coroutines.iscoroutine(coro), repr(coro)  # Not a coroutine function!
AssertionError: <Task pending coro=<async_update_ha_state() running at /usr/local/lib/python3.4/dist-packages/homeassistant-0.35.0.dev0-py3.4.egg/homeassistant/helpers/entity.py:191>>
```

--- a/homeassistant/components/alarm_control_panel/__init__.py
+++ b/homeassistant/components/alarm_control_panel/__init__.py
@@ -115,7 +115,7 @@ def async_setup(hass, config):
             update_coro = hass.loop.create_task(
                 alarm.async_update_ha_state(True))
             if hasattr(alarm, 'async_update'):
-                update_tasks.append(hass.loop.create_task(update_coro))
+                update_tasks.append(update_coro)
             else:
                 yield from update_coro
 

--- a/homeassistant/components/light/__init__.py
+++ b/homeassistant/components/light/__init__.py
@@ -253,7 +253,7 @@ def async_setup(hass, config):
             update_coro = hass.loop.create_task(
                 light.async_update_ha_state(True))
             if hasattr(light, 'async_update'):
-                update_tasks.append(hass.loop.create_task(update_coro))
+                update_tasks.append(update_coro)
             else:
                 yield from update_coro
 

--- a/homeassistant/components/remote/__init__.py
+++ b/homeassistant/components/remote/__init__.py
@@ -115,7 +115,7 @@ def async_setup(hass, config):
             update_coro = hass.loop.create_task(
                 remote.async_update_ha_state(True))
             if hasattr(remote, 'async_update'):
-                update_tasks.append(hass.loop.create_task(update_coro))
+                update_tasks.append(update_coro)
             else:
                 yield from update_coro
 

--- a/homeassistant/components/switch/__init__.py
+++ b/homeassistant/components/switch/__init__.py
@@ -98,7 +98,7 @@ def async_setup(hass, config):
             update_coro = hass.loop.create_task(
                 switch.async_update_ha_state(True))
             if hasattr(switch, 'async_update'):
-                update_tasks.append(hass.loop.create_task(update_coro))
+                update_tasks.append(update_coro)
             else:
                 yield from update_coro
 

